RIOT-OS/RIOT
# usbus: Unify terminology to use 'descriptor' everywhere

### Contribution description

This PR unifies the used terminology within USBUS to match the terminology from the specs.
`hdr`, `hdrs`, `headers`, etc is replaced everywhere with `descr` `descriptors` and `descriptors`.
I've also renamed some formatter functions to make it more consistent and hopefully a bit more clear.

Two very minimal once used functions have been merged with their calling functions. Let me know if it is too much to piggy-back here.

*note*: As some enums and struct members have been modified, this is a small API change.

### Testing procedure

Most of the PR is renaming of structs, documentation and function names. Impact should be minimal.

For the review, please verify that 'descriptor' and related words are used everywhere, also please look at some of the new enum names and function names and check if they make sense from a somewhat more outsider point of view. Mainly if the members in `usbus_descr_gen_funcs_t` have  clear names for the reviewer. Suggestions welcome.

Testing if the descriptors generated before and after this PR are identical should be enough for functional testing.

### Issues/PRs references

~~depends on #12429 ~~
